defi-pulse/src/index-allocations/index.ts

```diff
@@ -0,0 +1,45 @@
+import { ethers } from 'ethers'
+import { util } from '@chainlink/ea-bootstrap'
+
+type Allocations = {
+  components: string[]
+  units: number[]
+}
+
+/*
+  NOTICE!
+
+  The current implementation is fetching data directly from SetToken contracts (https://etherscan.io/address/0x78733fa5e70e3ab61dc49d93921b289e4b667093#code)
+  Note that this implementation won't work in other networks unless we deploy a copy of the contract.
+
+  The correct implementation should use SetProtocol.js typed library instead to fetch data directly from the SetToken contract directly. 
+  The ChainlinkAdapter.getAllocations(ISetToken _setToken) should be reimplemented in JS in order to use it.
+*/
+
+const ABI = [
+  {
+    inputs: [{ internalType: 'contract ISetToken', name: '_setToken', type: 'address' }],
+    name: 'getAllocations',
+    outputs: [
+      { internalType: 'address[]', name: '', type: 'address[]' },
+      { internalType: 'int256[]', name: '', type: 'int256[]' },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+]
+
+export const getAllocations = async (
+  contractAddress: string,
+  setAddress: string,
+): Promise<Allocations> => {
+  const rpcUrl = util.getRequiredEnv('RPC_URL')
```

Comment:
We should ingest this as a parameter so the config location is not coupled to this function. Then on the top lvl when we call this function we can set the `rpcUrl`:

```javascript
const rpcUrl = util.getRequiredEnv('RPC_URL')
const { components, units } = await getAllocations(asset.adapter, asset.address, rpcUrl)
```